src/server/dflycmd.cc

```diff
@@ -97,6 +97,20 @@ struct TransactionGuard {
 
   Transaction* t;
 };
+
+struct ExpirationGuard {
+  explicit ExpirationGuard(EngineShard* s = EngineShard::tlocal()) : shard(s) {
+    VLOG(2) << "Expiration guard";
+    shard->db_slice().SetExpireAllowed(false);
+  }
+
+  ~ExpirationGuard() {
```

Comment:
Createing Guard class for this works well though I personally would just directly call this function and will use absl::cleanup SetExpireAllowed(true)